pkg/agent/core/ngt/config/config_test.go

```diff
@@ -52,31 +56,278 @@ func TestNewConfig(t *testing.T) {
+					}
+					if err := f.Close(); err != nil {
+						t.Fatal(err)
+					}
+				},
+				afterFunc: func(t *testing.T, a args) {
+					t.Helper()
+					if err := os.Remove(a.path); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					wantCfg: &Data{
+						GlobalConfig: config.GlobalConfig{
+							Version: "v1.0.0",
+						},
+						Server: &config.Servers{
+							FullShutdownDuration: "10ms",
+							ShutdownStrategy:     make([]string, 0),
+							StartUpStrategy:      make([]string, 0),
+							TLS: &config.TLS{
+								Enabled: false,
+							},
+						},
+						Observability: &config.Observability{
+							Enabled: true,
+							Collector: &config.Collector{
+								Metrics: new(config.Metrics),
+							},
+							Trace:      new(config.Trace),
+							Prometheus: new(config.Prometheus),
+							Jaeger:     new(config.Jaeger),
+							Stackdriver: &config.Stackdriver{
+								Client:   new(config.StackdriverClient),
+								Exporter: new(config.StackdriverExporter),
+								Profiler: new(config.StackdriverProfiler),
+							},
+						},
+						NGT: &config.NGT{
+							IndexPath: "/var/index",
+							VQueue:    new(config.VQueue),
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			data := `
+                version: v1.0.0
+                server_config:
+                  full_shutdown_duration: 10ms
+                ngt:
+                  index_path: /var/index
+                `
+			return test{
+				name: "return Data and nil when the yaml bind successes but the input yaml value of observability is empty",
+				args: args{
+					path: "bind_success_but_observability_is_empty.yaml",
+				},
+				beforeFunc: func(t *testing.T, a args) {
+					t.Helper()
+					f, err := os.Create(a.path)
+					if err != nil {
+						t.Fatal(err)
+					}
+					if _, err := f.Write([]byte(data)); err != nil {
+						t.Fatal(err)
+					}
+					if err := f.Close(); err != nil {
+						t.Fatal(err)
+					}
+				},
+				afterFunc: func(t *testing.T, a args) {
+					t.Helper()
+					if err := os.Remove(a.path); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					wantCfg: &Data{
+						GlobalConfig: config.GlobalConfig{
+							Version: "v1.0.0",
+						},
+						Server: &config.Servers{
+							FullShutdownDuration: "10ms",
+							ShutdownStrategy:     make([]string, 0),
+							StartUpStrategy:      make([]string, 0),
+							TLS: &config.TLS{
+								Enabled: false,
+							},
+						},
+						Observability: new(config.Observability),
+						NGT: &config.NGT{
+							IndexPath: "/var/index",
+							VQueue:    new(config.VQueue),
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			path := "not_found.txt"
+			return test{
+				name: "test_case_2",
+				args: args{
+					path: path,
+				},
+				want: want{
+					wantCfg: nil,
+					err: &fs.PathError{
+						Op:   "open",
+						Path: path,
+						Err:  syscall.Errno(0x2),
+					},
+				},
+			}
+		}(),
 	}
```

Comment:
can you add more case? like empty config file, no read permission of config file.

Reply:
thank you
I fixed: https://github.com/vdaas/vald/pull/1219/commits/c3fcd2c20cf3ac721a26fc79abe9df5f6ff7380c